src/node.cc

```diff
@@ -448,20 +453,47 @@ void LoadEnvironment(Environment* env) {
 
 
 #ifdef __POSIX__
+
+static std::atomic<struct sigaction> previous_sigsegv_action;
+
+void TrapWebAssemblyOrContinue(int signo, siginfo_t* info, void* ucontext) {
+  if (!v8::TryHandleWebAssemblyTrapPosix(signo, info, ucontext)) {
+    auto prev = previous_sigsegv_action.load();
+    if (prev.sa_sigaction != nullptr) {
+      prev.sa_sigaction(signo, info, ucontext);
+    } else {
+      uv_tty_reset_mode();
+#ifdef __FreeBSD__
+      // FreeBSD has a nasty bug, see RegisterSignalHandler for details
```

Comment:
Not a blocker, just an observation, but I expect that's no longer relevant for the freebsd versions we support.

Reply:
@nodejs/platform-freebsd can anyone confirm or deny?

Reply:
@bnoordhuis do you know if there's any more background/detail on the issue?

@kostikbel any insight? The referenced comment is:
```
  // FreeBSD has a nasty bug with SA_RESETHAND reseting the SA_SIGINFO, that is
  // in turn set for a libthr wrapper. This leads to a crash.
  // Work around the issue by manually setting SIG_DFL in the signal handler
```

Reply:
From my copy of POSIX 1003.1-2017 AKA SuSv7:
> SA_RESETHAND If set, the disposition of the signal shall be reset to SIG_DFL and the SA_SIGINFO flag shall be cleared on entry to the signal handler.

And we do exactly that.  Am I reading right that this behavior causes some troubles, when combined with libthr ?  Can you give a minimal example ?

Reply:
@emaste @kostikbel The bug report is https://github.com/node-v0.x-archive/node/issues/9326, triage starts in https://github.com/nodejs/node-v0.x-archive/issues/9326#issuecomment-83848852.

I'm reasonably sure that bug was fixed in fbsd 11 but you probably know for sure, @kostikbel, since you authored the fix.

Reply:
Ok, so should this comment (and possibly some code) removed ?  Since I believe all supported FreeBSD versions have the SA_RESETHAND | SA_SIGINFO fix.

Reply:
That sounds about right (removing code & comment.) I've opened #27515 to track it.